scripts/deploy/github_steps.rb

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env ruby
+
+require 'open3'
+require 'octokit'
+
+require_relative 'common'
+
+def ensure_clean_repo()
+  repo_dir = File.basename(Dir.getwd)
+  if repo_dir != "stripe-android"
+    abort("You must run this script from 'stripe-android'.")
+  end
+
+  stdout, stderr, status = Open3.capture3("git status --porcelain")
+
+  if !stdout.empty?
+    abort("You must have a clean working directory to continue.")
+  end
+end
+
+def pull_latest()
+    execute_or_fail("git checkout #{@branch}")
+    execute_or_fail("git pull")
+end
+
+def switch_to_release_branch()
+    # Ensure that the branch doesn't already exist locally or remotely before creating it here.
+    delete_release_branch
+    execute_or_fail("git checkout -b #{release_branch}")
+end
+
+def delete_release_branch()
+    execute("git push origin --delete #{release_branch}")
+    execute("git branch -d #{release_branch}")
+end
+
+def create_version_bump_pr()
+    begin
+        execute_or_fail("git add README.md")
+        execute_or_fail("git add gradle.properties")
+        execute_or_fail("git add stripe-core/src/main/java/com/stripe/android/core/version/StripeSdkVersion.kt")
+        execute_or_fail("git commit -m \"Bump version to #{@version}\"")
+        execute_or_fail("git push -u origin")
+    rescue
+        # Undo all of the above if any of the steps fail.
+        execute("git reset HEAD~")
+        # Don't continue if any of the above failed.
+        raise
+    end
+
+
+    pr_description = create_pr_description()
+
+    options = {}
+    if (@is_dry_run)
+        options["draft"] = true
+    end
+
+    response = github_login.create_pull_request(
+        "stripe/stripe-android",
+        "master",
+        release_branch,
+        "Bump version to #{@version}",
+        pr_description,
+        options
+    )
+
+    rputs "Merge the version bump PR"
+    `open #{response.html_url}`
+    wait_for_user
+end
+
+def revert_all_changes()
+    execute_or_fail("git reset HEAD~")
+    execute_or_fail("git restore README.md")
+    execute_or_fail("git restore gradle.properties")
+    execute_or_fail("git restore stripe-core/src/main/java/com/stripe/android/core/version/StripeSdkVersion.kt")
+end
+
+private def release_branch
+    "release/#{@version}"
+end
+
+private def github_login
+  token = `fetch-password -q bindings/gh-tokens/$USER`
+  if $?.exitstatus != 0
+    puts "Couldn't fetch GitHub token. Follow the Android SDK Deploy Guide (https://go/android-sdk-deploy) to set up a token. \a".red
```

Comment:
```suggestion
    puts "Couldn't fetch GitHub token. Follow the Android SDK Deploy Guide (http://go/android-sdk-deploy) to set up a token. \a".red
```

Go doesn't support https.

Reply:
Ohh wait... It does now! Nice! nevermind.

Reply:
oop, nice! Reverted that change